pkg/grpc/grpc.go

```diff
@@ -0,0 +1,238 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	base "cosmossdk.io/api/cosmos/base/tendermint/v1beta1"
+
+	"github.com/archway-network/validator-exporter/pkg/config"
+	"github.com/archway-network/validator-exporter/pkg/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	slashing "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	staking "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"google.golang.org/grpc"
+
+	log "github.com/archway-network/validator-exporter/pkg/logger"
+)
+
+const valConsStr = "valcons"
+
+type Client struct {
+	ctx       context.Context
```

Comment:
I think context should not be in the struct, reference:
https://go.dev/blog/context-and-structs